src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -20,7 +20,7 @@ import { TopNavMenuItem } from './top_nav_menu_item';
 export type TopNavMenuProps = StatefulSearchBarProps &
   Omit<SearchBarProps, 'kibana' | 'intl' | 'timeHistory'> & {
     config?: TopNavMenuData[];
-    badges?: Array<EuiBadgeProps & { badgeText: string }>;
+    badges?: Array<EuiBadgeProps & { badgeText: string, toolTipProps: { content: string; position: ToolTipPositions }}>;
```

Comment:
```suggestion
    badges?: Array<EuiBadgeProps & { badgeText: string, toolTipProps: EuiToolTipProps }>;
```

Rather than using a new type here for `{ content: string; position: ToolTipPositions }` you could use the existing `EuiToolTipProps` here to support passing any tooltip prop through. This makes it easier to use (less casting required), and means that it will support any tooltip prop. I left another comment to describe how you could use it.

Reply:
@ThomThomson ,
Thanks for your suggestion! I have replaced `ToolTipPositions `with `EuiToolTipProps`.